java/src/edu/wpi/first/wpilibj/networktables/NetworkTable.java

```diff
@@ -852,8 +852,8 @@ public boolean putRaw(String key, ByteBuffer value, int len) {
   public boolean putValue(String key, Object value) throws IllegalArgumentException {
     if (value instanceof Boolean)
       return NetworkTablesJNI.putBoolean(pathWithSep + key, ((Boolean)value).booleanValue());
-    else if (value instanceof Double)
-      return NetworkTablesJNI.putDouble(pathWithSep + key, ((Double)value).doubleValue());
+    else if (value instanceof Number && !(value instanceof Long))
```

Comment:
Not sure if this is the way we want to do this. BigDecimal, for example, is a Number. If we're excluding Long, we should exclude everything that doesn't fit. Doing this with casts seems fundamentally broken in terms of object oriented languages. Number has a `.doubleValue()`. We should just call that, and document that rounding can/will occur as Numbers are sent as doubles.